examples/pytorch/text-generation/run_generation.py

```diff
@@ -382,14 +394,25 @@ def main():
         input_ids = encoded_prompt
 
     if args.jit:
-        jit_input_texts = ["jit"]
+        jit_input_texts = ["enable jit"]
         jit_inputs = prepare_jit_inputs(jit_input_texts, model, tokenizer)
         torch._C._jit_set_texpr_fuser_enabled(False)
         model.config.return_dict = False
-        traced_model = torch.jit.trace(model, jit_inputs, strict=False)
-        traced_model = torch.jit.freeze(traced_model.eval())
-        traced_model(*jit_inputs)
-        traced_model(*jit_inputs)
+        if version.parse(version.parse(torch.__version__).base_version) >= version.parse("2.1.0"):
+            traced_model = torch.jit.trace(model, example_kwarg_inputs=dict(jit_inputs), strict=False)
+            traced_model = torch.jit.freeze(traced_model.eval())
+            traced_model(**jit_inputs)
+            traced_model(**jit_inputs)
+        else:
+            if hasattr(model, "forward"):
+                sig = inspect.signature(model.forward)
+            else:
+                sig = inspect.signature(model.__call__)
+            jit_inputs = tuple(jit_inputs[key] for key in sig.parameters if jit_inputs.get(key, None) is not None)
+            traced_model = torch.jit.trace(model, jit_inputs, strict=False)
+            traced_model = torch.jit.freeze(traced_model.eval())
+            traced_model(*jit_inputs)
+            traced_model(*jit_inputs)
```

Comment:
Hi, @amyeroberts , thanks for the review,  I have a fix for jit.trace with dict input which needs be used in text generation case. the fix is https://github.com/pytorch/pytorch/pull/97866, contained in pytorch nightly version(2.1.0.xxx). 2.0 version does not have this fix. Without the fix, we could not use jit.trace with dict input for text generation. it will fail caused by the past_key_value. As for tuple input for jit.trace. it has some limitation, such as there should not be any parameter appeared in the middle of jit_inputs variables in the forward() func, like LlamaForCausalLM
position_id in forward(), appeared before past_key_value. which will make the jit.trace confused and fail. in the nightly pytorch version. jit.trace for mostly model will success with the dict input. That's why I add version check here.

Reply:
As mentioned above - the examples should be simple demonstrations. As nightly builds are inherently unstable it makes this code brittle. We can either:
* Remove the torch 2.1.0 logic logic branch here
* Wait until 2.1.0 is released, remove the other branch and specify the torch version in the example requirements